src/abstraction/ek.rs

```diff
@@ -0,0 +1,171 @@
+        0x69, 0xaa, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00,
+    ];
+
+    match alg {
+        AsymmetricAlgorithm::Rsa => Tpm2BPublicBuilder::new()
+            .with_type(TPM2_ALG_RSA)
+            .with_name_alg(TPM2_ALG_SHA256)
+            .with_object_attributes(obj_attrs)
+            .with_auth_policy(32, authpolicy)
+            .with_parms(PublicParmsUnion::RsaDetail(TPMS_RSA_PARMS {
+                symmetric: TPMT_SYM_DEF_OBJECT {
+                    algorithm: TPM2_ALG_AES,
+                    keyBits: TPMU_SYM_KEY_BITS { aes: 128 },
+                    mode: TPMU_SYM_MODE { aes: TPM2_ALG_CFB },
+                },
+                scheme: TPMT_RSA_SCHEME {
+                    scheme: TPM2_ALG_NULL,
+                    details: Default::default(),
+                },
+                keyBits: 2048,
+                exponent: 0,
+            }))
+            .with_unique(PublicIdUnion::Rsa(Box::new(TPM2B_PUBLIC_KEY_RSA {
+                size: 256,
+                buffer: [0; 512],
+            })))
+            .build(),
+        AsymmetricAlgorithm::Ecc => Tpm2BPublicBuilder::new()
+            .with_type(TPM2_ALG_ECC)
+            .with_name_alg(TPM2_ALG_SHA256)
+            .with_object_attributes(obj_attrs)
+            .with_auth_policy(32, authpolicy)
+            .with_parms(PublicParmsUnion::EccDetail(TPMS_ECC_PARMS {
+                symmetric: TPMT_SYM_DEF_OBJECT {
+                    algorithm: TPM2_ALG_AES,
+                    keyBits: TPMU_SYM_KEY_BITS { sym: 128 },
+                    mode: TPMU_SYM_MODE { sym: TPM2_ALG_CFB },
+                },
+                scheme: TPMT_ECC_SCHEME {
+                    scheme: TPM2_ALG_NULL,
+                    details: TPMU_ASYM_SCHEME {
+                        anySig: TPMS_SCHEME_HASH {
+                            hashAlg: TPM2_ALG_NULL,
+                        },
+                    },
+                },
+                curveID: TPM2_ECC_NIST_P256,
+                kdf: TPMT_KDF_SCHEME {
+                    scheme: TPM2_ALG_NULL,
+                    details: Default::default(),
+                },
+            }))
+            .with_unique(PublicIdUnion::Ecc(Box::new(TPMS_ECC_POINT {
+                x: TPM2B_ECC_PARAMETER {
+                    size: 32,
+                    buffer: [0; 128],
+                },
+                y: TPM2B_ECC_PARAMETER {
+                    size: 32,
+                    buffer: [0; 128],
+                },
+            })))
+            .build(),
+    }
+}
+
+/// Create the Endorsement Key object from the specification templates
+pub fn create_ek_object(context: &mut Context, alg: AsymmetricAlgorithm) -> Result<KeyHandle> {
+    let ek_public = create_ek_public_from_default_template(alg)?;
+
+    let ses = create_nullauth_session(context)?;
```

Comment:
I see in the function below you flush the context of the `nullauth` session - is there no need to do that here?

Reply:
Yep, that was needed. I just added the `execute_with_nullauth_session` to `Context`, since there may be more cases where you need to run something that's `TPM_ST_SESSIONS` without actually needing to do authentication.